src/tests/ut/visit/visit_test.cpp

```diff
@@ -2,47 +2,63 @@
 
 #include "visit_test.hpp"
 
+#include <algorithm>
+
 namespace
 {
-TEST_F(AssociationTest, ExtentDemo)
+TEST_F(VisitTestsFixture, ExtentDemo)
 {
     auto content = Visit::getExtent().size();
     auto expected = 3;
     EXPECT_EQ(content, expected);
 }
 
-TEST_F(AssociationTest, AssociationDemo)
+TEST_F(VisitTestsFixture, AssociationDemo)
 {
     auto content = visit1->getDoctorAssociation()->getLastName();
     auto expected = "Tracz";
     EXPECT_EQ(content, expected);
 }
 
-TEST_F(AssociationTest, AssociationDemo2)
+TEST_F(VisitTestsFixture, AssociationDemo2)
 {
     auto content = doc2->getVisitAssociations().begin()->get()->getVisitInformation();
     auto expected = "Tworze klub ninja!";
     EXPECT_EQ(content, expected);
 }
 
-TEST_F(AssociationTest, AssociationDemo3)
+TEST_F(VisitTestsFixture, AssociationDemo3)
 {
     std::string content;
     auto associations = doc2->getVisitAssociations();
-    if (auto it = associations.begin(); it != associations.end())
-    {
-        std::advance(it, 1);
-        if (it != doc2->getVisitAssociations().end())
-        {
-            content = it->get()->getVisitInformation();
-        }
-        else
-        {
-            content = "null";
-        }
-    }
+
     auto expected = "I w klubie sa sami fajni ludzie";
-    EXPECT_EQ(content, expected);
+
+    EXPECT_TRUE(std::ranges::any_of(
+        associations, [&expected](const auto& visitSPtr) { return visitSPtr->getVisitInformation() == expected; }));
+}
+
+TEST_F(VisitTestsFixture, UpdateTreatmentsShouldReplacePreviouslyStoredTreatments)
+{
+    auto visitOneTreatmentsBefore = visit1->getTreatments();
+    auto visitTwoTreatmentsBefore = visit2->getTreatments();
+    auto visitThreeTreatmentsBefore = visit3->getTreatments();
+    // initial asserts
+    ASSERT_TRUE(visitOneTreatmentsBefore.empty());
+    ASSERT_EQ(visitTwoTreatmentsBefore, std::vector{Treatment::TeethWhitening});
+    ASSERT_EQ(visitThreeTreatmentsBefore,
+              (std::vector{Treatment::TeethCleaning, Treatment::RootCanal, Treatment::ToothExtraction}));
+
+    // update treatments for visit
```

Comment:
same here
and consider creating a method in fixture to do such updates

Reply:
Single update function implemented in test fixture.
Comment removed.